src/SDK/Common/Retry/RetryPolicyInterface.php

```diff
@@ -0,0 +1,131 @@
+ *          backoffMultiplier,  - multiplier value used to calculate the delay before next attempt
+ *          jitter,             - jitter value to add randomness to delay value
+ *          retryableStatusCodes- array of retryable status code. All the status codes apart from this will not be retried
+ *      )
+ *
+ *    User can set the retryable status codes separately as well using following api
+ *    $this->setRetryableStatusCodes([
+ *      \Grpc\STATUS_CANCELLED,
+ *      \Grpc\STATUS_DEADLINE_EXCEEDED,
+ *      \Grpc\STATUS_PERMISSION_DENIED,
+ *      \Grpc\STATUS_RESOURCE_EXHAUSTED,
+ *      \Grpc\STATUS_ABORTED,
+ *      \Grpc\STATUS_OUT_OF_RANGE,
+ *      \Grpc\STATUS_UNAVAILABLE,
+ *      \Grpc\STATUS_DATA_LOSS,
+ *      \Grpc\STATUS_UNAUTHENTICATED,
+ *      ]);
+ *
+ *    User can also provide the delay scheduler with following api:
+ *    $this->setDelayScheduler(new NonBlockingScheduler());
+ *    This api will define whether delay function is blocking or non blocking
+ *    NonBlockingScheduler will be more useful when fiber is implemented.
+ *    By Default scheduler is of type BlockingScheduler.
+ */
+interface RetryPolicyInterface
+{
+    public const DEFAULT_MAX_ATTEMPTS = 5;
+    public const DEFAULT_INITIAL_BACKOFF = 1;
+    public const DEFAULT_MAX_BACKOFF = 10;
+    public const DEFAULT_BACKOFF_MULTIPLIER = 1.5;
+    public const DEFAULT_JITTER = 0.1;
+
+    /**
+     * Returns whether the request should be retried.
+     *
+     * @param int $attempt - current number of retry attempt
+     * @param int $status - SpanExporterStatusInterface
+     * @param RetryableExportException|null $exception: exception caught while retrying
+     *
+     * @return bool
+     */
+    public function shouldRetry(
+        int $attempt,
+        int $status,
+        ?RetryableExportException $exception
+    ): ?bool;
+
+    /**
+     * Returns the time to wait in seconds.
+     * @param int $attempt: current number of retry attempt
+     *
+     * @return int
+     */
+    public function getDelay(int $attempt): int;
+
+    public static function getDefault(): RetryPolicyInterface;
+
+    public function getMaxAttempts(): int;
+
+    public function setMaxAttempts(int $max_attempts);
+
+    public function getInitialBackoff(): float;
+
+    public function setInitialBackoff(float $initial_backoff);
+
+    public function getMaxBackoff(): int;
+
+    public function setMaxBackoff(int $max_backoff);
+
+    public function getBackoffMultiplier(): float;
+
+    public function setBackoffMultipler(float $backoff_multiplier);
+
+    public function getJitter(): float;
+
+    public function setJitter(float $jitter);
+
+    public function setRetryableStatusCodes(array $statusCodes);
+
+    public function getRetryableStatusCodes(): array;
+
+    public function setDelayScheduler(SchedulerInterface $scheduler);
+
+    public function getDelayScheduler(): SchedulerInterface;
```

Comment:
Shouldn't be part of the retry policy, unless the retry policy performs the waiting instead of returning the delay.

Reply:
I have kept it here with the idea that each policy must have a delayschedular (blocking delay scheduler is the default type). so its better to keep in the policyInterface. we get the delayscheduler instance in exporter class using retryPolicy instance and use its delay function in exporter itself. So that way retryPolicy does perform the delay

Reply:
I have removed all setter getters from RetryPolicyInterface and moved them in actual retry policy